### ISSUE TITLE
loadProjectFromDirectory file sync, import & export sync

### ISSUE THREAD

**Issue by samuelstroschein** [opral/inlang-sdk#159]
### Context

@samuel.stroschein said in [MESDK-176](https://linear.app/opral/issue/MESDK-176/addunpackfile-to-enable-collaboration-between-dev-tools-web-apps#comment-48894e60):

> Update: `saveProjectToDirectory` is not needed until we need roundtrip capabilities from dev tools <-> web apps. As long as the dev tools are single player, `loadProjectFromDirectory` can set up file watching and auto in- and export.
>
> cc @felix.haeberle

### Task

- [X] add file syncing between host <-> lix
- [ ] add watching for changes in `toBeImportedFiles()`
- [ ] add auto import/export

**Comment by samuelstroschein** [comment]
@martin.lysk1 can you take this over? 

progress is in branch [https://github.com/opral/monorepo/tree/mesdk-200-loadprojectfromdirectory](https://github.com/opral/monorepo/tree/mesdk-200-loadprojectfromdirectory)

[Understanding File Syncing in Inline SDK](https://www.loom.com/share/8c1d38dff17748b5a3a7b06e3fd454b5?sid=971b527c-6a89-4d5c-bc63-a07817270107)

**Comment by martin-lysk** [comment]
sure i am on it

**Comment by martin-lysk** [comment]
Quick question about syncing from the inlang lix to the host:<br>Process is like this:

1. create a project in memory
2. import everything from the inlang folder passed in start watching - on change on 
3. on change on files in lix, copy the files to the host.

Regarding 3. some followup questions.

How shall we distinguish between files we want to sync to the host (like settings) and those we don't want to pass to the host, like the `inlang.db` i guess.<br><br>**Option 1** - hardcode a block list and just skip the `inlang.db`<br>**Option 2** - keep track of files imported via copyFilesFromDiskRecursive and sync only those that we imported to the host again. 

Lets  go for Option 1 @samuel.stroschein?

How shall we notice a change in a file

Option 1 - should we add a callback to the lix api - on file change, create a trigger in lix?Option 2 - should we create a trigger using the lix db object in inlang

I prefere Option 1 but 2 is faster

Both will break with  [https://linear.app/opral/issue/LIXDK-102/re-visit-simplifying-the-change-queue-implementation#comment-65eb3485](https://linear.app/opral/issue/LIXDK-102/re-visit-simplifying-the-change-queue-implementation#comment-65eb3485)

**Comment by samuelstroschein** [comment]
> How shall we distinguish between files we want to sync to the host (like settings) and those we don't want to pass to the host, like the `inlang.db` i guess.

Yes, hardcode ignored files. 

> How shall we notice a change in a file

You can use `pollQuery` which only emits an event if the underlying query result changes. 

(`pollQuery` is simulating an observable for sqlite. Granted, not real time but good enough. Lix won't need callbacks and manually setting up triggers is also not needed)

**Comment by samuelstroschein** [comment]
@martin.lysk1 I completed [MESDK-176](https://linear.app/opral/issue/MESDK-176/add-saveprojectfileasdirectory-to-enable-collaboration-between-dev). During the implementation the following observations emerged that you should consider for this issue: 

1. Don't forget to use `absolutePath` and `withAbsolutePaths` when you interact with fs stuff that plugins use. See [https://github.com/opral/inlang-sdk/issues/174](https://github.com/opral/inlang-sdk/issues/174) 
2. the `saveProjectToDirectory` tests might get flaky if `loadProjectFromDirectory` establishes watchers. To mitigate the problem I propose to add a `watch: boolean` flag to `loadProjectFromDirectory` that turns the sync on or off.

**Comment by samuelstroschein** [comment]
After a sync with @martin.lysk1 we concluded: 

* using lix [LIXDK-126](https://linear.app/opral/issue/LIXDK-126/branching) to solve the sync issues makes sense
* because branching is WIP, martin will work on [LIXDK-147](https://linear.app/opral/issue/LIXDK-147/change-discussions) and [LIXDK-158](https://linear.app/opral/issue/LIXDK-158/introduce-changetag-to-filter-history-and-replace-the-purpose-of) in the meantime

**Comment by martin-lysk** [comment]
Quick update from my side:<br><br>Blocked by [MESDK-186](https://linear.app/opral/issue/MESDK-186/settings-under-change-control) compare [https://discord.com/channels/897438559458430986/1287684940418514964/1287700196381167656](https://discord.com/channels/897438559458430986/1287684940418514964/1287700196381167656)

**Comment by samuelstroschein** [comment]
marking as on hold because martin is focusing on a csv hosted demo for the onsite
